### PR TITLE
Infer current branch checkout, remove repoHead arguments

### PR DESCRIPTION
**Pull request by atrakh**
This PR removes the `repoHead` command line argument in effort to simplify the utility. 

Added new logic to infer the branch name using a special `git rev-parse` flag.

### CODE REVIEW

File: pkg/parse/parse.go [hunk 1]
```diff
@@ -64,29 +64,20 @@ func Parse() {
...
-	// 	if err != nil {
-	// 		fatal("Failed to create temp directory for repo clone", err)
-	// 	}
-	// 	defer os.RemoveAll(dir)
-	// 	cmd.Workspace = dir
-	// 	err = cmd.Clone(endpoint)
-	// 	if err != nil {
-	// 		fatal("Unable to clone repo", err)
-	// 	}
-	// }
```
Comment (atrakh) [inline pkg/parse/parse.go]: Decided it's pretty safe to remove this commented out code path. If we ever want this functionality back, we can do a git revert on the commit that commented this code out.

File: pkg/parse/parse.go [hunk 2]
```diff
@@ -64,29 +64,20 @@ func Parse() {
...
+	cmd := git.Git{Workspace: o.Dir.Value()}
 
-	headSha, err := cmd.RevParse()
+	currBranch, err := cmd.BranchName()
+	if err != nil {
+		log.Error.Fatalf("error parsing git branch name: %s", err)
+	} else if strings.ToUpper(currBranch) == "HEAD" || currBranch == "" {
+		log.Error.Fatalf("error parsing git branch name: git repo at %s must be checked out to a valid branch", cmd.Workspace)
```
Comment (atrakh) [inline pkg/parse/parse.go]: This errors out if you actually named your branch `head`, but I don't think that's a huge issue

File: internal/git/git.go [hunk 3]
```diff
@@ -9,8 +9,7 @@ import (
 
 type Git struct {
 	Workspace string
-	Head      string
-	RepoName  string
+	// Head      string
```
Comment (vlymar) [inline internal/git/git.go]: why is this commented out?

File: pkg/parse/parse.go [hunk 4]
```diff
@@ -177,12 +168,7 @@ func getFlags(ldApi ld.ApiClient) ([]string, error) {
 }
 
 func (b *branch) findReferences(cmd git.Git, flags []string, ctxLines int, exclude *regexp.Regexp) (grepResultLines, error) {
-	err := cmd.Checkout()
```
Comment (vlymar) [inline pkg/parse/parse.go]: sanity check - do github/bitbucket give us the repo with the right branch already checked out?

File: pkg/parse/parse.go (hunk 2 above)
Comment (vlymar) [reply pkg/parse/parse.go]: why is a branch name of `head` a problem?

File: build/package/circleci/orb.yml [hunk 5]
```diff
@@ -51,7 +51,6 @@ commands:
               -repoType=${LD_REPO_TYPE} \
               -repoUrl=${LD_REPO_URL} \
               -repoName=${CIRCLE_PROJECT_REPONAME} \
-              -repoHead=${CIRCLE_BRANCH} \
```
Comment (vlymar) [inline build/package/circleci/orb.yml]: we can assume that in CLI situations people will manually have the desired branch checked out, and hopefully github/bitbucket pre-checkout the correct branch, but does circle provide the repo with the correct branch checked out?

Comment (atrakh) [reply build/package/circleci/orb.yml]: Yes, circle checks out to the correct branch.

File: pkg/parse/parse.go (hunk 4 above)
Comment (atrakh) [reply pkg/parse/parse.go]: Yes, I checked bb and github, and they appear to give us the correct branch checkout. I'll make a release candidate out of this branch and double check as well.

File: pkg/parse/parse.go (hunk 2 above)
Comment (atrakh) [reply pkg/parse/parse.go]: It's the output returned by `git rev-parse --abbrev-ref` if you're in a detached state. I suppose we could detect this as a problem earlier on so this check doesn't happen in `parse.go`

Comment (vlymar) [reply pkg/parse/parse.go]: yeah is there some way to just make sure the repo isn't in a detached state? that being said, i guess this is a nice to have. I could just see a user being surprised that we'd have this restriction. We can also always add support for branches named `head` later w/o a backwards incompatible change

**Comment by atrakh** [comment]
filed: https://app.clubhouse.io/launchdarkly/story/30379/ld-find-code-refs-add-extra-checks-for-detached-state-to-support-branches-named-head

**Pull request merged**